uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/manager-leave/components/manager-leave-requests_test.js

```diff
@@ -1,30 +1,899 @@
+        $provide = _$provide_;
+      }));
+
+      beforeEach(inject(function (AbsencePeriodAPIMock, AbsenceTypeAPIMock, LeaveRequestAPIMock) {
+        $provide.value('AbsencePeriodAPI', AbsencePeriodAPIMock);
+        $provide.value('AbsenceTypeAPI', AbsenceTypeAPIMock);
+        $provide.value('LeaveRequestAPI', LeaveRequestAPIMock);
+      }));
+
+      beforeEach(inject(['api.contact.mock', function (_ContactAPIMock_) {
+        ContactAPIMock = _ContactAPIMock_;
+      }]));
+
+      beforeEach(inject(function (_$compile_, _$log_, _$rootScope_, _$q_, _OptionGroup_, _OptionGroupAPIMock_,
+                                  _AbsencePeriod_, _AbsenceType_, _LeaveRequest_, _Contact_) {
         $compile = _$compile_;
         $log = _$log_;
+        $q = _$q_;
         $rootScope = _$rootScope_;
+        OptionGroupAPIMock = _OptionGroupAPIMock_;
+        OptionGroup = _OptionGroup_;
+        AbsenceType = _AbsenceType_;
+        AbsencePeriod = _AbsencePeriod_;
+        LeaveRequest = _LeaveRequest_;
+        Contact = _Contact_
+      }));
 
+      beforeEach(function () {
         spyOn($log, 'debug');
 
+        spyOn(AbsencePeriod, 'all').and.callThrough();
+        spyOn(AbsenceType, 'all').and.callThrough();
+
+        spyOn(Contact, 'all').and.callFake(function () {
+          return $q.resolve(ContactAPIMock.mockedContacts());
+        });
+
+        spyOn(OptionGroup, 'valuesOf').and.callFake(function (name) {
+          return OptionGroupAPIMock.valuesOf(name);
+        });
+      });
+
+      beforeEach(function () {
         compileComponent();
-      }));
+      });
 
       it('is initialized', function () {
         expect($log.debug).toHaveBeenCalled();
       });
 
+      describe('initially', function () {
+
+        it('the filter section is closed', function () {
+          expect(controller.isFilterExpanded).toBe(false);
+        });
+
+        describe('pagination', function () {
+
+          it('is set to page 1', function () {
+            expect(controller.pagination.page).toBe(1);
+          });
+
+          it('has page size set to 7', function () {
+            expect(controller.pagination.size).toBe(7);
+          });
+        });
+      });
+
+      describe('data loading::', function () {
+
+        //TODO need to figure out how to test variables which are changed when controller gets initialized
+        xdescribe('before loading starts', function () {
+
+          it('loading should be hidden', function () {
+
+          });
+
+          it('leave requests are empty', function () {
+
+          });
+
+          it('absencePeriods are empty', function () {
+
+          });
+
+          it('absenceTypes are empty', function () {
+
+          });
+
+          it('statusCount is reset', function () {
+
+          });
+        });
+
+        describe('after data loading is complete', function () {
+
+          describe('loading', function () {
+
+            it('should be hidden for page', function () {
```

Comment:
never use "should" in `it` titles. The mood is always imperative
```js
it('is hidden for page', function () {});
it('is hidden for content', function () {});
```
This applies for all other titles you have in the test

By the way the title is not clear. What is hidden? It reads "after data loading is complete / loading / it should be hidden for page", but what's hidden? Can you rephrase it? Do you mean that the loading is complete for page and content?